File: batches_c/batch_14_completely_different_logic/compare.c
#include <stdio.h>

void selection_sort(int *arr, int len) {
    for (int i = 0; i < len; i++) {
        int min_idx = i;
        for (int j = i + 1; j < len; j++) {
            if (arr[j] < arr[min_idx]) min_idx = j;
        }
        int tmp = arr[i]; arr[i] = arr[min_idx]; arr[min_idx] = tmp;
    }
}

int main() {
    int data[] = {64, 34, 25, 12, 22, 11, 90};
    selection_sort(data, 7);
    for (int i = 0; i < 7; i++) printf("%d ", data[i]);
    printf("\n");
    return 0;
}

File: batches_csharp/batch_14_completely_different_logic/compare.cs
using System;

class SelectionSort {
    static void Sort(int[] arr) {
        for (int i = 0; i < arr.Length; i++) {
            int minIdx = i;
            for (int j = i + 1; j < arr.Length; j++)
                if (arr[j] < arr[minIdx]) minIdx = j;
            int tmp = arr[i]; arr[i] = arr[minIdx]; arr[minIdx] = tmp;
        }
    }

    static void Main() {
        int[] data = {64, 34, 25, 12, 22, 11, 90};
        Sort(data);
        Console.WriteLine(string.Join(", ", data));
    }
}
